lib/practice_exercises.rb

```diff
@@ -1,13 +1,42 @@
-
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: 0(n), If the string gets longer, then the loop will take longer to execute.
+# Space Complexity: 0(n), The downcase method is duplicating the string.
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  string = string.downcase.gsub(/[^a-z0-9]/i,"")
+  if string.length == 0
+    return true
+  end
+
+  low_value = 0
+  high_value = string.length - 1
+  while (low_value <= high_value) do
+    if string[low_value] != string[high_value]
+      return false
+    else
+      low_value += 1
+      high_value  -= 1
+    end
+  end
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: ???
 def longest_prefix(strings)
```

Comment:
👍 Your space complexity is going to be O(m) where m is the length of the shortest string (because you're building the prefix).  

You are also looping through 2 things, the 1st string and the array of strings.  So the time complexity is O(nm)